samples/excel/85-preview-apis/direct-precedents.yaml

```diff
@@ -0,0 +1,154 @@
+order: 6
+id: excel-direct-precedents
+name: Direct precedents
+description: 'This sample shows how to highlight precedent cells which directly affect the value of the currently selected cell, using the direct precedents API.'
+host: EXCEL
+api_set:
+    ExcelAPI: '1.12'
+script:
+    content: |
+        $("#setup").click(() => tryCatch(setup));
+        $("#select-first-cell").click(() => tryCatch(selectFirstCell));
+        $("#select-second-cell").click(() => tryCatch(selectSecondCell));
+        $("#get-direct-precedents").click(() => tryCatch(getDirectPrecedents));
+        $("#clear-highlighting").click(() => tryCatch(setup));
```

Comment:
Let's write a separate method to clear the formatting. I think it'll be more clear to the people reading the sample, plus it might help teach someone how to clear formatting. You can get the used data ranges for each worksheet (`Worksheet.getUsedRange()`) and then clear the fill on both (Range.format.fill.clear()).

Happy to write this up in Script Lab for you, if you'd like.

Reply:
Ok, would love to see how you write this. I considered writing a new method to clear the formatting at the time, but it seemed unnecessary. I agree with your concerns/suggestions though.